PORTFOLIO: Added Mobile Menu JS Function

peterdtitan opened the pull request:
Used query selectors to implement a mobile menu that also closes when a menu item is clicked. The following files were added/modified:

- script.js was added
- index.html was modified
- style.css was modified
- linters were added for js

<img width="514" alt="Screenshot 2022-06-14 at 6 58 14 PM 1" src="https://user-images.githubusercontent.com/70844072/173622817-e90eecc5-24dd-464a-b24a-42fe15eb3fb4.png">

Commit:
Add Linters

Commit:
Update mobile menu

Commit:
Update mobile menu

Commit:
Fix Linter errors

Commit:
Fix Linter errors

tufoinnkuo10 commented on index.html:
```diff
@@ -28,7 +28,7 @@
                     <li class="nav-item">
                         <a href="#">Contact</a>
                     </li>
-                    <li class="nav-item">
+                    <li id="hidemail" class="nav-item">
                         <a href="#"><img src="./assets/Disabled.png" alt="mail"></a>
                     </li>
```
- [x] links work well great job. please use the same Figma font weight and size, also the line breaks under the link
- your version
![Screenshot (121)](https://user-images.githubusercontent.com/65298048/173632051-b72e501a-9ee8-43d0-b923-5938ff29a792.png)
- Figma version
![Screenshot (116)](https://user-images.githubusercontent.com/65298048/173632205-045f2f1e-6978-4778-bc73-f7e2dfc9b938.png)

peterdtitan replied:
Updated

tufoinnkuo10 commented on index.html:
```diff
@@ -28,7 +28,7 @@
                     <li class="nav-item">
                         <a href="#">Contact</a>
                     </li>
-                    <li class="nav-item">
```
- [x] please fix the popup in such a way that when it comes up the user cant see what you are doing in the background.
![Screenshot (122)_LI](https://user-images.githubusercontent.com/65298048/173633089-44a1331d-5f24-42fb-998e-64bff96acecd.jpg)

peterdtitan replied:
Fixed popup and background activity

tufoinnkuo10 commented on index.html:
```diff
@@ -28,7 +28,7 @@
                     </li>
-                    <li class="nav-item">
+                    <li id="hidemail" class="nav-item">
                         <a href="#"><img src="./assets/Disabled.png" alt="mail"></a>
```
- [x] please ensure that the close button is with the mobile popup menu and not separated from it. when scrolling on your mobile menu the close button separates from the other links.
your version 👇🏾
![Screenshot (123)_LI](https://user-images.githubusercontent.com/65298048/173634017-2b9e2869-cd37-40e6-8eea-0f7fd4bf59c6.jpg)

peterdtitan replied:
Made Changes to keep the hamburger menu icon fixed.

tufoinnkuo10 reviewed (changes requested):
Hi @peterdtitan 
There are some issues that you still need to work on to go to the next project but you are almost there!

To Highlight
✅ You used the correct Github flow
✅ Descriptive pull request message
✅ linter error checks validated
✅ Documented your work professionally
✅ Mobile menu hamburger works as expected
✅ Menu close icon works perfectly well

Required Changes ♻️
Check the comments under the review

![Naah](https://media.giphy.com/media/Jq824R93JsLwZCaiSL/giphy.gif)


Optional suggestions
_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise.
As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using this [form.](https://airtable.com/shrQAqnBwek5a0O0s)

Commit:
Update Styling

Commit:
Update Styling

peterdtitan commented:
Changed appropriate styling to fit the Figma template.

<img width="549" alt="Screenshot 2022-06-15 at 8 29 00 AM" src="https://user-images.githubusercontent.com/70844072/173747744-55430f91-4d6a-4479-bf15-c388664c65ca.png">

sja-thedude commented on script.js:
```diff
@@ -0,0 +1,17 @@
+const hamburger = document.querySelector('.hamburger');
+const navMenu = document.querySelector('.nav-menu');
+const logo = document.querySelector('.logo');
+const navItem = document.querySelector('.nav-item');
+
+hamburger.addEventListener('click', () => {
+  hamburger.classList.toggle('active');
+  navMenu.classList.toggle('active');
+  logo.classList.toggle('active');
+  navItem.classList.toggle('active');
+});
+
+document.querySelectorAll('.nav-item').forEach((n) => n.addEventListener('click', () => {
+  hamburger.classList.remove('active');
+  navMenu.classList.remove('active');
+  logo.classList.remove('active');
+}));
```
- [ ] Great job so far, Kindly make sure to add margin to the top and on the right side as shown below.
- [ ] Kindly make sure that the line after all menu options is of the same width as per the Figma design.
- [ ] Kindly make sure to add space after all menu options and before the line as shown.
- [ ] Optional: Kindly follow the same colour for the close button as per the design.

https://user-images.githubusercontent.com/70269507/173751663-a4e17f6a-99e7-41b6-ad4c-59a881e31ff9.mov

peterdtitan replied:
Updated styling. Matched template.

sja-thedude reviewed (changes requested):
## Required Changes ♻️

Hi @peterdtitan, This is @sja-thedude your code reviewer.

Great work on making the changes requested by a previous reviewer 👏 
You've done well implementing some of the requested changes, but there are still some which aren't addressed yet.

![200w](https://user-images.githubusercontent.com/70269507/159484303-3187dacc-4a82-4263-bb1a-8d72417ce1d9.gif)

 ## Highlights
✔️ Great job on making the changes asked previously.

 ## Required Changes ♻️
- [ ] _Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me [@sja-thedude] in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

Commit:
Update Styling

Commit:
Update Styling

peterdtitan commented:
Updated the styling to match the Figma template. Added margins and spaced link separators in the menu.

<img width="529" alt="Screenshot 2022-06-15 at 9 37 54 AM" src="https://user-images.githubusercontent.com/70844072/173759861-5a204ef3-0715-4b2d-a060-1f4ccd97cda6.png">

koachcode reviewed (approved):
Hi @peterdtitan , 

# STATUS: APPROVED 🟢

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 

## TO HIGHLIGHT: 
- Your project is professional 💯 
- Your repo is professional 💯 
- Well structured files 💯 
- You set up linter files and no errors. 💯 

![well-1](https://user-images.githubusercontent.com/64173189/128382030-38e7de91-f6f2-45cf-b54e-8068e0446e4c.gif)

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
